src/openlifu/db/database.py

```diff
@@ -208,6 +208,30 @@ def write_transducer(self, transducer, on_conflict: OnConflictOpts=OnConflictOpt
                 raise ValueError("Invalid 'on_conflict' option. Use 'error', 'overwrite', or 'skip'.")
 
         transducer_filename = self.get_transducer_filename(transducer_id)
+        transducer_parent_dir = transducer_filename.parent
+        transducer_parent_dir.mkdir(exist_ok = True)
+
+        # Copy the transducer data files to database.
+        # If tranducer data files not provided, check that any files previously
+        # associated with the transducer object exist.
+        if registration_surface_model_filepath:
+            if not Path(registration_surface_model_filepath).exists():
```

Comment:
setting `registration_surface_model_filepath = Path(registration_surface_model_filepath)` earlier on can help avoid having to repeatedly call the constructor `Path()` below. similar for `transducer_body_model_filepath`